### PR TITLE
Update to Facebook airlift

### PR DESCRIPTION
**Pull request by wenleix**
Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftConnectorFactory.java [hunk 1]
```diff
@@ -79,10 +79,6 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
 
             return injector.getInstance(ThriftConnector.class);
         }
-        catch (InterruptedException ie) {
```
Comment (highker) [inline presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftConnectorFactory.java]: Why these changes?

Comment (wenleix) [reply presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftConnectorFactory.java]: @highker : Later version of `Boostrap` no longer  throw this exception. In fact we should backport https://github.com/prestosql/presto/pull/1549 to remove even Exception handling.

File: pom.xml [hunk 2]
```diff
@@ -876,6 +883,65 @@
                 <groupId>io.airlift.discovery</groupId>
                 <artifactId>discovery-server</artifactId>
                 <version>1.29</version>
+                <!-- TODO: fork io.airlift.discovery:discovery-server and change the dependency there -->
```
[the remark this answers is not in the document]
Comment (wenleix) [reply pom.xml]: @iceted : No we don't need to. `io.airlift.discovery:discovery-server` also depends on `io.airlift` so it needs to be forked.

[the remark this answers is not in the document]
Comment (iceted) [reply pom.xml]: What's wrong with it depending on io.airlift? It's in a different namespace so it doesn't conflict

**Comment by wenleix** [comment]
Superseded by https://github.com/prestodb/presto/pull/13549
